src/main/java/org/hydev/mcpm/client/interaction/ProgressSpeedCalculator.java

```diff
@@ -0,0 +1,53 @@
+package org.hydev.mcpm.client.interaction;
+
+import org.hydev.mcpm.utils.Pair;
+
+import java.util.ArrayDeque;
+import java.util.Queue;
+
+/**
+ * An implementation of ProgressSpeedCalculator using a queue and sliding window method.
+ *
```

Comment:
May be beneficial to include a more substantial docstring explaining what the ProgressSpeedCalculator actually does lol